bundles/org.openhab.binding.haywardomnilogic/src/main/java/org/openhab/binding/haywardomnilogic/internal/handler/HaywardBridgeHandler.java

```diff
@@ -0,0 +1,458 @@
+                .header(HttpHeader.ACCEPT_ENCODING, "gzip, deflate").version(HttpVersion.HTTP_1_1)
+                .header(HttpHeader.CONNECTION, "keep-alive").header(HttpHeader.HOST, "www.haywardomnilogic.com:80")
+                .timeout(10, TimeUnit.SECONDS);
+    }
+
+    public synchronized String httpXmlResponse(String urlParameters) throws HaywardException {
+        String urlParameterslength = Integer.toString(urlParameters.length());
+        String statusMessage;
+
+        try {
+            ContentResponse httpResponse = sendRequestBuilder(config.endpointUrl, HttpMethod.POST)
+                    .content(new StringContentProvider(urlParameters), "text/xml; charset=utf-8")
+                    .header(HttpHeader.CONTENT_LENGTH, urlParameterslength).send();
+
+            int status = httpResponse.getStatus();
+            String xmlResponse = httpResponse.getContentAsString();
+
+            List<String> statusMessages = evaluateXPath("/Response/Parameters//Parameter[@name='StatusMessage']/text()",
+                    xmlResponse);
+            if (!(statusMessages.isEmpty())) {
+                statusMessage = statusMessages.get(0);
+            } else {
+                statusMessage = httpResponse.getReason();
+            }
+
+            if (status == 200) {
+                if (logger.isTraceEnabled()) {
+                    logger.trace("Hayward Connection thing:  {} Hayward http command: {}", getCallingMethod(),
+                            urlParameters);
+                    logger.trace("Hayward Connection thing:  {} Hayward http response: {} {}", getCallingMethod(),
+                            statusMessage, xmlResponse);
+                }
+                return xmlResponse;
+            } else {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Hayward Connection thing:  {} Hayward http command: {}", getCallingMethod(),
+                            urlParameters);
+                    logger.debug("Hayward Connection thing:  {} Hayward http response: {}", getCallingMethod(), status);
+                }
+                return "";
+            }
+        } catch (ExecutionException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Unable to resolve host.  Check Hayward hostname and your internet connection.");
+            return "";
+        } catch (TimeoutException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Connection Timeout.  Check Hayward hostname and your internet connection.");
+            return "";
+        } catch (InterruptedException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Connection Timeout.  Check Hayward hostname and your internet connection.");
+            return "";
+        }
```

Comment:
This exception is thrown when OH is shut-down. That means the current operation should be finished as fast as possible and no further operations should be done. You can achieve this by not catching the exception here but throw the `InterruptedException` and catch it at the highest level you can.

Reply:
Removed.

Reply:
Most of the Hayward Thing Handlers handleCommand methods call the bridge.httpXmlResponse.  The handleCommand methods all ultimately override the ThingHandler.handleCommand method.  I'm not seeing a way to throw an exception on an override which means I can only throw this up one level, but then would need to catch it in most of the thing handler handle command methods.  Let me know if I'm missing something obvious.  Thanks.

Reply:
You're right, the exception can't be thrown any higher than in `handleCommand()`. That should be sufficient in your case as you can exit `handlerCommand()` immediately if you catch `InterruptedException` there.